FlingEngine/Gameplay/inc/Camera.h

```diff
@@ -0,0 +1,79 @@
+#pragma once
+#include "FlingMath.h"
+
+namespace Fling
+{
+	/**
+	 * @brief Base class for camera, meant to be overridden
+	 * 
+	 */
+	class Camera
+	{
+	public:
+		Camera() :
+			m_nearPlane(0.1f),
+			m_farPlane(1000.0f),
+			m_fieldOfView(glm::radians(45.0f)),
+			m_aspectRatio(1.6180f) //golden ratio
+		{
+		}
+
+		virtual ~Camera() = default;
+		virtual void Update(float dt) = 0;
+
+		/**
+		 * @brief Gets the near plane of the view frustrum
+		 * 
+		 * @return const float& m_nearPlane
+		 */
+		const float& GetNearPlane() const { return m_nearPlane; }
```

Comment:
Returning a const ref isn't great here because this is just a float. Makes the return value actually 8 bytes (cause its an address now, so on x64 thats 8 bytes) and a float is just 4 bytes